pkg/environment/api/organization.go

```diff
@@ -323,6 +337,230 @@ func (s *EnvironmentService) validateCreateOrganizationRequest(
 	return nil
 }
 
+func (s *EnvironmentService) createOrganizationNoCommand(
+	ctx context.Context,
+	req *environmentproto.CreateOrganizationRequest,
+	editor *eventproto.Editor,
+	localizer locale.Localizer,
+) (*environmentproto.CreateOrganizationResponse, error) {
+	if err := s.validateCreateOrganizationRequestNoCommand(req, localizer); err != nil {
+		return nil, err
+	}
+	// Create the organization
+	name := strings.TrimSpace(req.Name)
+	urlCode := strings.TrimSpace(req.UrlCode)
+	organization, err := domain.NewOrganization(
+		name,
+		urlCode,
+		req.OwnerEmail,
+		req.Description,
+		req.IsTrial,
+		req.IsSystemAdmin,
+	)
+	if err != nil {
+		s.logger.Error(
+			"Failed to create an organization",
+			log.FieldsFromImcomingContext(ctx).AddFields(
+				zap.Error(err),
+			)...,
+		)
+		return nil, statusInternal.Err()
+	}
+	// Begin the SQL transaction
+	tx, err := s.mysqlClient.BeginTx(ctx)
+	if err != nil {
+		s.logger.Error(
+			"Failed to begin transaction",
+			log.FieldsFromImcomingContext(ctx).AddFields(
+				zap.Error(err),
+			)...,
+		)
+		dt, err := statusInternal.WithDetails(&errdetails.LocalizedMessage{
+			Locale:  localizer.GetLocale(),
+			Message: localizer.MustLocalize(locale.InternalServerError),
+		})
+		if err != nil {
+			return nil, statusInternal.Err()
+		}
+		return nil, dt.Err()
+	}
+	err = s.mysqlClient.RunInTransaction(ctx, tx, func() error {
+		orgStorage := v2es.NewOrganizationStorage(tx)
+		if organization.Organization.SystemAdmin {
+			org, err := orgStorage.GetSystemAdminOrganization(ctx)
+			if err != nil {
+				return err
+			}
+			if org != nil {
+				return v2es.ErrOrganizationAlreadyExists
+			}
+		}
+		if err := orgStorage.CreateOrganization(ctx, organization); err != nil {
+			return err
+		}
+		// Create a default project
+		project, err := s.createDefaultProject(
+			ctx,
+			tx,
+			organization.Id,
+			organization.OwnerEmail,
+		)
+		if err != nil {
+			s.logger.Error(
+				"Failed to create the default project",
+				log.FieldsFromImcomingContext(ctx).AddFields(
+					zap.Error(err),
+				)...,
+			)
+			return err
+		}
+		// Create default environments
+		envRoles, err := s.createDefaultEnvironments(
+			ctx,
+			tx,
+			organization.Id,
+			project,
+		)
+		if err != nil {
+			s.logger.Error(
+				"Failed to create the default environments",
+				log.FieldsFromImcomingContext(ctx).AddFields(
+					zap.Error(err),
+				)...,
+			)
+			return err
+		}
+		// Create the admin account using the environment roles created in the last step
+		// Because the account storage has a different implementation,
+		// we can't create the account using the same transaction.
```

Comment:
As I commented here, the account storage has a different implementation, so I can't reuse the same transaction to create a new account. When we refactor the organization storage layer, I'll improve this.